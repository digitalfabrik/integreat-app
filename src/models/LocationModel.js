// @flow

class LocationModel {
  _name: ?string
  _address: ?string
  _town: ?string
  _state: ?string
  _postcode: ?string
  _region: ?string
  _country: ?string
  _latitude: ?string
  _longitude: ?string

  constructor ({ name, address, town, state, postcode, region, country, latitude, longitude }: {|
                 name: ?string, address: ?string, town: ?string, state: ?string, postcode: ?string, region: ?string,
                 country: ?string, latitude?: ?string, longitude?: ?string
               |}
  ) {
    this._name = name
    this._address = address
    this._town = town
    this._state = state
    this._postcode = postcode
    this._region = region
    this._country = country
    this._latitude = latitude
    this._longitude = longitude
    this._name = name
  }

  get name (): ?string {
    return this._name
  }

  get address (): ?string {
    return this._address
  }

  get town (): ?string {
    return this._town
  }

  get state (): ?string {
    return this._state
  }

  get postcode (): ?string {
    return this._postcode
  }

  get region (): ?string {
    return this._region
  }

  get country (): ?string {
    return this._country
  }

  get longitude (): ?string {
    return this._longitude
  }

  get latitude (): ?string {
    return this._latitude
  }

  get location (): ?string {
    const town = this._postcode && this._town ? `${this._postcode} ${this._town}` : this._town
    if (!town && !this._address && !this._name) {
      return null
    }
    return [this._name, this._address, town]
      .filter(value => !!value)
      .join(', ')
  }
}

export default LocationModel
