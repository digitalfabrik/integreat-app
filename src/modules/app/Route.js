import UrlPattern from 'url-pattern'

/**
 * Holds data about an url and allows maching it
 */
class Route {
  id
  path
  pattern

  constructor ({id = null, path}) {
    this.id = id
    this.path = path

    this.pattern = new UrlPattern(path)
  }

  /**
   * Creates the path for this route with supplied params
   *
   * @see url-pattern package for more information
   * @param params The params which will be used to build the path
   */
  stringify (params = {}) {
    return this.pattern.stringify(params)
  }

  hasPath (path) {
    return this.path === path
  }
}

export default Route
