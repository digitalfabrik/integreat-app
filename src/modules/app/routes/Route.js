// @flow

import type { Node } from 'react'
import type { Route as RouterRouteType } from 'redux-first-router'
import type { AllPayloadsType } from './types'

class Route<T, P> {
  _name: string
  _getRoutePath: P => string
  _renderPage: T => Node
  _route: RouterRouteType
  _getRequiredPayloads: AllPayloadsType => T

  constructor ({ name, getRoutePath, getLanguageChangeAction, renderPage, route, getRequiredPayloads }: {|
    name: string, getRoutePath: P => string, renderPage: T => Node, route: RouterRouteType,
    getRequiredPayloads: AllPayloadsType => T
  |}) {
    this._name = name
    this._getRoutePath = getRoutePath
    this._renderPage = renderPage
    this._route = route
    this._getRequiredPayloads = getRequiredPayloads
  }

  get name (): string {
    return this._name
  }

  get getRoutePath (): P => string {
    return this._getRoutePath
  }

  get renderPage (): T => Node {
    return this._renderPage
  }

  get route (): RouterRouteType {
    return this._route
  }

  get getRequiredPayloads (): AllPayloadsType => T {
    return this._getRequiredPayloads
  }
}

export default Route
