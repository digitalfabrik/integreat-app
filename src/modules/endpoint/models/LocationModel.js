// @flow

class LocationModel {
  _address: ?string
  _town: ?string
  _postcode: ?string
  _latitude: ?string
  _longitude: ?string

  constructor ({address, town, postcode, latitude, longitude}: {|address: string, town: string, postcode: string,
    latitude: string, longitude: string|}) {
    this._address = address
    this._town = town
    this._postcode = postcode
    this._latitude = latitude
    this._longitude = longitude
  }

  get address (): ?string {
    return this._address
  }

  get town (): ?string {
    return this._town
  }

  get postcode (): ?string {
    return this._postcode
  }

  get longitude (): ?string {
    return this._longitude
  }

  get latitude (): ?string {
    return this._latitude
  }

  get location (): string {
    return `${this._address}, ${this._postcode} ${this._town}`
  }
}

export default LocationModel
