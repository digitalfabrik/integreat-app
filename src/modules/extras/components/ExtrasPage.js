// @flow

import * as React from 'react'
import TileModel from '../../common/models/TileModel'
import { ExtraModel } from '@integreat-app/integreat-api-client'
import ContentNotFoundError from '../../common/errors/ContentNotFoundError'
import FailureSwitcher from '../../common/components/FailureSwitcher'
import Tiles from '../../common/components/Tiles'
import type { TFunction } from 'react-i18next'

type RouteParamsType = {|city: string, language: string, offerHash?: string|}
type SprungbrettRouteParamsType = {|city: string, language: string|}
type PropsType = {|
  city: string,
  language: string,
  extras: Array<ExtraModel>,
  extraId: ?string,
  navigateToCategories: (path: string) => void,
  t: TFunction
|}

export const SPRUNGBRETT_ROUTE = 'SPRUNGBRETT'
export const SPRUNGBRETT_EXTRA = 'sprungbrett'

export const WOHNEN_ROUTE = 'WOHNEN'
export const WOHNEN_EXTRA = 'wohnen'

const getWohnenRoutePath = ({city, language, offerHash}: RouteParamsType): string =>
  `/${city}/${language}/extras/${WOHNEN_EXTRA}${offerHash ? `/${offerHash}` : ''}`
const getSprungbrettRoutePath = ({city, language}: SprungbrettRouteParamsType): string =>
  `/${city}/${language}/extras/${SPRUNGBRETT_EXTRA}`

export class ExtrasPage extends React.Component<PropsType> {
  onTilePress = (tile: TileModel) => {
    this.props.navigateToCategories(tile.path)
  }

  toTileModels (extras: Array<ExtraModel>): Array<TileModel> {
    const { city, language } = this.props
    return extras.map(
      extra => {
        let path = extra.path
        if (extra.alias === SPRUNGBRETT_EXTRA) {
          path = getSprungbrettRoutePath({city, language})
        } else if (extra.alias === WOHNEN_EXTRA) {
          path = getWohnenRoutePath({city, language})

        }

        return new TileModel({
          id: extra.alias,
          title: extra.title,
          // the url stored in the sprungbrett extra is the url of the endpoint
          path: path,
          thumbnail: extra.thumbnail,
          // every extra except from the sprungbrett extra is just a link to an external site
          isExternalUrl: path === extra.path,
          postData: extra.postData
        })
      }
    )
  }

  render () {
    const { city, extras, extraId, language, t } = this.props

    if (extraId) {
      // If there is an extraId, the route is invalid, because every internal extra has a separate route
      const error = new ContentNotFoundError({type: 'extra', id: extraId, city: city, language})
      return <FailureSwitcher error={error} />
    }

    return (
      <Tiles title={t('extras')} tiles={this.toTileModels(extras)} onTilePress={this.onTilePress} />
    )
  }
}