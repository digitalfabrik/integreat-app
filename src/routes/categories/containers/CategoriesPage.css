.breadcrumb-spacing {
  margin-bottom: 20px;
}

.toolbar {

}

.content {
  max-width: 1000px;
  box-sizing: border-box;
  flex-grow: 1;
  margin: 0 auto;
  padding: 0 16px;
  word-wrap: break-word;
}
