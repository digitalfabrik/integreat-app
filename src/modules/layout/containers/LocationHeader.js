// @flow

import React from 'react'
import type { Element } from 'react'
import { translate } from 'react-i18next'
import compose from 'lodash/fp/compose'

import LanguageSelector from '../../common/containers/LanguageSelector'
import searchIcon from '../assets/magnifier.svg'
import landingIcon from '../assets/location-icon.svg'
import Header from 'modules/layout/components/Header'
import HeaderNavigationItem from '../components/HeaderNavigationItem'
import HeaderActionItem from '../HeaderActionItem'
import { EXTRAS_ROUTE, goToExtras } from '../../app/routes/extras'
import { CATEGORIES_ROUTE, goToCategories } from '../../app/routes/categories'
import { EVENTS_ROUTE, goToEvents } from '../../app/routes/events'
import { goToSearch } from '../../app/routes/search'
import { goToLanding } from '../../app/routes/landing'
import { connect } from 'react-redux'

import type { LocationState } from 'redux-first-router/dist/flow-types'
import EventModel from '../../endpoint/models/EventModel'
import type { I18nTranslate, State } from '../../../flowTypes'

type Props = {
  events: ?Array<EventModel>,
  location: LocationState,
  viewportSmall: boolean,
  t: I18nTranslate,
  isEventsEnabled: boolean,
  isExtrasEnabled: boolean,
  onStickyTopChanged: number => {}
}

export class LocationHeader extends React.Component<Props> {
  getActionItems (): Array<HeaderActionItem> {
    const { location } = this.props
    const { city, language } = location.payload
    return [
      new HeaderActionItem({
        href: goToSearch(city, language),
        iconSrc: searchIcon
      }),
      new HeaderActionItem({
        href: goToLanding(language),
        iconSrc: landingIcon
      }),
      new HeaderActionItem({ node: <LanguageSelector isHeaderActionItem /> })
    ]
  }

  getNavigationItems (): Array<Element<typeof HeaderNavigationItem>> {
    const { t, isEventsEnabled, isExtrasEnabled, location, events } = this.props
    const { city, language } = location.payload
    const currentRoute = location.type

    const isEventsActive = events ? events.length > 0 : false
    const isCategoriesEnabled = isExtrasEnabled || isEventsEnabled

    const items: Array<Element<typeof HeaderNavigationItem>> = []

    if (isExtrasEnabled) {
      items.push(
        <HeaderNavigationItem
          key='extras'
          href={goToExtras(city, language)}
          selected={currentRoute === EXTRAS_ROUTE}
          text={t('extras')}
          active
        />
      )
    }

    if (isCategoriesEnabled) {
      items.push(
        <HeaderNavigationItem
          key='categories'
          href={goToCategories(city, language)}
          selected={currentRoute === CATEGORIES_ROUTE}
          text={t('categories')}
          active
        />
      )
    }

    if (isEventsEnabled) {
      items.push(
        <HeaderNavigationItem
          key='events'
          href={goToEvents(city, language)}
          selected={currentRoute === EVENTS_ROUTE}
          text={t('news')}
          tooltip={t('noNews')}
          active={isEventsActive}
        />
      )
    }

    return items
  }

  render () {
    const { city, language } = this.props.location.payload

    return (
      <Header
        viewportSmall={this.props.viewportSmall}
        logoHref={goToCategories(city, language)}
        actionItems={this.getActionItems()}
        navigationItems={this.getNavigationItems()}
        onStickyTopChanged={this.props.onStickyTopChanged}
      />
    )
  }
}

const mapStateToProps = (state: State) => ({
  location: state.location,
  viewportSmall: state.viewport.is.small,
  languages: state.languages.data,
  events: state.events.data
})

export default compose(connect(mapStateToProps), translate('layout'))(
  LocationHeader
)
