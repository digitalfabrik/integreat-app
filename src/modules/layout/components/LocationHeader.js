import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'

import LanguageSelector from '../../common/containers/LanguageSelector'
import searchIcon from '../assets/magnifier.svg'
import locationIcon from '../assets/location-icon.svg'
import languageIcon from '../assets/language-icon.svg'
import LocationModel from 'modules/endpoint/models/LocationModel'
import Header from 'modules/layout/components/Header'
import HeaderNavigationItem from '../HeaderNavigationItem'
import HeaderActionItem from '../HeaderActionItem'
import SearchPage from 'routes/search/containers/SearchPage'
import LandingPage from 'routes/landing/containers/LandingPage'
import CategoriesPage from 'routes/categories/containers/CategoriesPage'
import EventsPage from 'routes/events/containers/EventsPage'
import ExtrasPage from 'routes/extras/containers/ExtrasPage'

class LocationHeader extends React.Component {
  static propTypes = {
    matchRoute: PropTypes.func.isRequired,
    locationModel: PropTypes.instanceOf(LocationModel).isRequired,
    language: PropTypes.string.isRequired,
    currentPath: PropTypes.string.isRequired,
    viewportSmall: PropTypes.bool.isRequired,
    eventCount: PropTypes.number.isRequired,
    t: PropTypes.func.isRequired
  }

  getActionItems () {
    const {matchRoute} = this.props
    const currentParams = this.getCurrentParams()
    return [
      new HeaderActionItem({href: matchRoute(SearchPage).stringify(currentParams), iconSrc: searchIcon}),
      new HeaderActionItem({href: matchRoute(LandingPage).stringify(currentParams), iconSrc: locationIcon}),
      new HeaderActionItem({dropDownNode: <LanguageSelector />, iconSrc: languageIcon})
    ]
  }

  getCurrentParams () {
    return {
      location: this.props.locationModel.code,
      language: this.props.language
    }
  }

  getNavigationItems () {
    const {t, matchRoute, currentPath} = this.props
    const currentParams = this.getCurrentParams()

    const isEventsEnabled = () => this.props.locationModel.eventsEnabled
    const isExtrasEnabled = () => this.props.locationModel.extrasEnabled
    const isCategoriesEnabled = () => isExtrasEnabled() || isEventsEnabled()

    const isExtrasSelected = () => matchRoute(ExtrasPage).hasPath(currentPath)
    const isCategoriesSelected = () => matchRoute(CategoriesPage).hasPath(currentPath)
    const isEventsSelected = () => matchRoute(EventsPage).hasPath(currentPath)

    const isEventsActive = () => this.props.eventCount > 0

    const extras = isExtrasEnabled() &&
      new HeaderNavigationItem({
        href: matchRoute(ExtrasPage).stringify(currentParams),
        selected: isExtrasSelected(),
        text: t('extras'),
        active: true
      })

    const categories = isCategoriesEnabled() &&
      new HeaderNavigationItem({
        href: matchRoute(CategoriesPage).stringify(currentParams),
        selected: isCategoriesSelected(),
        text: t('categories'),
        active: true
      })

    const events = isEventsEnabled() &&
      new HeaderNavigationItem({
        href: matchRoute(EventsPage).stringify(currentParams),
        selected: isEventsSelected(),
        text: t('news'),
        active: isEventsActive(),
        tooltip: t('noNews')
      })

    return [extras, categories, events].filter(isEnabled => isEnabled)
  }

  render () {
    const {matchRoute} = this.props
    return <Header
      viewportSmall={this.props.viewportSmall}
      logoHref={matchRoute(CategoriesPage).stringify(this.getCurrentParams())}
      actionItems={this.getActionItems()}
      navigationItems={this.getNavigationItems()} />
  }
}

export default translate('layout')(LocationHeader)
