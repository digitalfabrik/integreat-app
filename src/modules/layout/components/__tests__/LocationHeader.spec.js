import { shallow } from 'enzyme'
import React from 'react'
import LocationHeader from '../LocationHeader'
import LocationModel from 'modules/endpoint/models/LocationModel'
import Route from '../../../app/Route'
import CategoriesPage from '../../../../routes/categories/containers/CategoriesPage'
import LandingPage from '../../../../routes/landing/containers/LandingPage'
import SearchPage from '../../../../routes/search/containers/SearchPage'
import EventsPage from '../../../../routes/events/containers/EventsPage'
import ExtrasPage from '../../../../routes/extras/containers/ExtrasPage'

describe('LocationHeader', () => {
  const matchRoute = id => {
    switch (id) {
      case CategoriesPage:
        return new Route({id, path: '/:location/:language(/*)'})
      case LandingPage:
        return new Route({id, path: '/:language'})
      case SearchPage:
        return new Route({id, path: '/:location/:language/search'})
      case EventsPage:
        return new Route({id, path: '/:location/:language/events(/:id)'})
      case ExtrasPage:
        return new Route({id, path: '/:location/:language/extras(/:extra)'})
    }
    throw new Error(`Route ${id} not found!`)
  }

  const language = 'de'

  const createLocation = (extrasEnabled, eventsEnabled) => new LocationModel({
    name: 'Mambo Nr. 5',
    code: 'location1',
    eventsEnabled,
    extrasEnabled
  })

  describe('NavigationItems', () => {
    it('should be empty, if extras and news are both disabled', () => {
      const component = shallow(<LocationHeader matchRoute={matchRoute} language={language}
                                                locationModel={createLocation(false, false)}
                                                currentPath='' />)
      expect(component.dive().prop('navigationItems')).toMatchSnapshot()
    })

    it('should show categories, if extras or news are enabled', () => {
      const extrasComp = shallow(<LocationHeader matchRoute={matchRoute} language={language}
                                                 locationModel={createLocation(true, false)}
                                                 currentPath='' />)
      const eventsComp = shallow(<LocationHeader matchRoute={matchRoute} language={language}
                                                 locationModel={createLocation(false, true)}
                                                 currentPath='' />)

      expect(extrasComp.dive().prop('navigationItems')).toMatchSnapshot()
      expect(eventsComp.dive().prop('navigationItems')).toMatchSnapshot()
    })

    it('should show extras, categories, events in this order', () => {
      const component = shallow(<LocationHeader matchRoute={matchRoute} language={language}
                                                locationModel={createLocation(true, true)}
                                                currentPath='' />)
      expect(component.dive().prop('navigationItems')).toMatchSnapshot()
    })

    it('should highlight categories if route corresponds', () => {
      const component = shallow(<LocationHeader matchRoute={matchRoute} language={language}
                                                locationModel={createLocation(true, true)}
                                                currentPath='/:location/:language(/*)' />)
      expect(component.dive().prop('navigationItems')).toMatchSnapshot()
    })

    it('should highlight events if route corresponds', () => {
      const component = shallow(<LocationHeader matchRoute={matchRoute} language={language}
                                                locationModel={createLocation(true, true)}
                                                currentPath='/:location/:language/events(/:id)' />)

      expect(component.dive().prop('navigationItems')).toMatchSnapshot()
    })

    it('should highlight extras if extras route is selected', () => {
      const component = shallow(<LocationHeader matchRoute={matchRoute} language={language}
                                                locationModel={createLocation(true, true)}
                                                route='/:location/:language/extras(/:extra)' />)

      expect(component.dive().prop('navigationItems')).toMatchSnapshot()
    })
  })

  it('should match snapshot', () => {
    const component = shallow(<LocationHeader matchRoute={matchRoute} language={language}
                                              locationModel={createLocation(true, true)} path='' />)
    expect(component.dive()).toMatchSnapshot()
  })
})
