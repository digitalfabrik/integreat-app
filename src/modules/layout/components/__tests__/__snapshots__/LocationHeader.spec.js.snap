// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationHeader NavigationItems should be empty, if extras and news are both disabled 1`] = `Array []`;

exports[`LocationHeader NavigationItems should highlight categories if route corresponds 1`] = `
Array [
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de/extras",
    "_selected": false,
    "_text": "layout:extras",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de",
    "_selected": true,
    "_text": "layout:categories",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": false,
    "_href": "/location1/de/events",
    "_selected": false,
    "_text": "layout:news",
    "_tooltip": "layout:noNews",
  },
]
`;

exports[`LocationHeader NavigationItems should highlight events if route corresponds 1`] = `
Array [
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de/extras",
    "_selected": false,
    "_text": "layout:extras",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de",
    "_selected": false,
    "_text": "layout:categories",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": false,
    "_href": "/location1/de/events",
    "_selected": true,
    "_text": "layout:news",
    "_tooltip": "layout:noNews",
  },
]
`;

exports[`LocationHeader NavigationItems should highlight extras if extras route is selected 1`] = `
Array [
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de/extras",
    "_selected": false,
    "_text": "layout:extras",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de",
    "_selected": false,
    "_text": "layout:categories",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": false,
    "_href": "/location1/de/events",
    "_selected": false,
    "_text": "layout:news",
    "_tooltip": "layout:noNews",
  },
]
`;

exports[`LocationHeader NavigationItems should show categories, if extras or news are enabled 1`] = `
Array [
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de/extras",
    "_selected": false,
    "_text": "layout:extras",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de",
    "_selected": false,
    "_text": "layout:categories",
    "_tooltip": undefined,
  },
]
`;

exports[`LocationHeader NavigationItems should show categories, if extras or news are enabled 2`] = `
Array [
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de",
    "_selected": false,
    "_text": "layout:categories",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": false,
    "_href": "/location1/de/events",
    "_selected": false,
    "_text": "layout:news",
    "_tooltip": "layout:noNews",
  },
]
`;

exports[`LocationHeader NavigationItems should show extras, categories, events in this order 1`] = `
Array [
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de/extras",
    "_selected": false,
    "_text": "layout:extras",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": true,
    "_href": "/location1/de",
    "_selected": false,
    "_text": "layout:categories",
    "_tooltip": undefined,
  },
  HeaderNavigationItem {
    "_active": false,
    "_href": "/location1/de/events",
    "_selected": false,
    "_text": "layout:news",
    "_tooltip": "layout:noNews",
  },
]
`;

exports[`LocationHeader should match snapshot 1`] = `
<Header
  actionItems={
    Array [
      HeaderActionItem {
        "_dropDownNode": undefined,
        "_href": "/location1/de/search",
        "_iconSrc": "test-file-stub",
      },
      HeaderActionItem {
        "_dropDownNode": undefined,
        "_href": "/de",
        "_iconSrc": "test-file-stub",
      },
      HeaderActionItem {
        "_dropDownNode": <Connect(getContext(Connect(languagesFetcher))) />,
        "_href": undefined,
        "_iconSrc": "test-file-stub",
      },
    ]
  }
  logoHref="/location1/de"
  navigationItems={
    Array [
      HeaderNavigationItem {
        "_active": true,
        "_href": "/location1/de/extras",
        "_selected": false,
        "_text": "layout:extras",
        "_tooltip": undefined,
      },
      HeaderNavigationItem {
        "_active": true,
        "_href": "/location1/de",
        "_selected": false,
        "_text": "layout:categories",
        "_tooltip": undefined,
      },
      HeaderNavigationItem {
        "_active": false,
        "_href": "/location1/de/events",
        "_selected": false,
        "_text": "layout:news",
        "_tooltip": "layout:noNews",
      },
    ]
  }
/>
`;
